ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/transactional/TransactionalInterceptorRequiresNew.java

```diff
@@ -49,7 +49,7 @@ public Object intercept(InvocationContext ic) throws Exception {
 
     @Override
     protected Object doIntercept(TransactionManager tm, Transaction tx, InvocationContext ic) throws Exception {
-        if (tx != null) {
+        if (tx != null && ic != null) {
```

Comment:
So if I understand correctly then this fix expects that the tx could not be `null` - i.e., in case of EJB manages  transactions and the transactions was started - but the `ic` is null as it's EJB that is the manager and is not the responsibility of CDI to work here.
Do I interpret this right?

Reply:
Yes, I got you. Whenever EJB manages the transactions the value of ic could be null. I have missed that scenario, will change it to as it was before.